### PR TITLE
Add a build step which validates each source file for the inclusion of our license header

### PR DESCRIPTION
**Pull request by rastersize**
This PR adds a build step which validates that all source files contains our license header. The script did find a few files which where missing it :smile: 

It also adds Travis-CI group folding to our different steps, to make the output pretties and easier to find. 

Credits for the nice little `diff` command goes to @dflems :tada:

### CODE REVIEW

File: ci/script.sh [hunk 1]
```diff
@@ -1,14 +1,29 @@
...
     -sdk "$TEST_SDK" \
     -destination "$TEST_DEST" \
     $EXTRA_ARGUMENTS \
-         | xcpretty -c -f `xcpretty-travis-formatter`
+        | xcpretty -c -f `xcpretty-travis-formatter`
+
+
+# License conformance
+travis_fold_open "License conformance"
+./ci/validate_license_conformance.sh {include/SPTDataLoader/*.h,SPTDataLoader/*.{h,m}}
```
Comment (rastersize) [inline ci/script.sh]: I put the license validation after the general build actions. However as they run so fast, and they will fail the build, should we have the check before building and test? Wondering if that would make the fail/fix cycle quicker.

Downside to that is that you’d have to fix the license issue (trivial) before you’re even told about potential code failures.

**Comment by 8W9aG** [comment]
I'm for putting it before the build, it seems like it is better to fail fast

**Comment by rastersize** [comment]
@8W9aG Ok, I moved it.

**Comment by rastersize** [comment]
This is ready for review @8W9aG @jgavris @dflems :smiley:

**Comment by jgavris** [comment]
Awesome efforts! I am mostly against merging 'revert' and 'fixup' commits into mainline from a topic branch. How do others feel about squashing into the winner commits?

**Comment by 8W9aG** [comment]
I think each to their own on this issue, their are valid arguments for both

**Comment by rastersize** [comment]
@jgavris @8W9aG @dflems Ready for review!

@jgavris I went ahead and squashed the commits a bit. It should be less noisy now. However I left the commits which do separate things separate. Also some commits would have been annoying to squash (specifically squashing 5d0cafa4707c84949fb2bfe1a4faaffc04f1a83f into 838963f9307facb417963ec82d376c574c218885) so I left those as-is.

**Comment by 8W9aG** [comment]
:+1: Very nice

**Comment by 8W9aG** [comment]
:+1:

**Comment by jgavris** [comment]
:+1:

**Pull request merged**